pkg/apis/admission/types.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AdmissionControlConfiguration is the cluster-wide configuration of default
+// initializers and external admission webhooks that are applied to each
+// resource.
+type AdmissionControlConfiguration struct {
+	metav1.TypeMeta
+
+	// ResourceInitializers is a list of resources and their default initializers
+	ResourceInitializers []ResouceInitializer
+
+	// ExternalAdmissionHooks is a list of external admission webhooks and the
+	// affected resources and operations.
+	ExternalAdmissionHooks []ExternalAdmissionHook
+}
+
+// ResouceInitializer describes the default initializers that will be
+// applied to a resource. The order of initializers is sensitive.
+type ResouceInitializer struct {
+	// APIGroup is the API group of the resource
+	APIGroup string
+
+	// APIVersions is the API Versions of the resource
+	// '*' means all API Versions.
+	// If '*' is present, the length of the slice must be one.
+	// Defaults to '*'.
+	APIVersions []string
+
+	// Resource is resource to be initialized
+	Resource string
+
+	// Initializers is a list of initializers that will be applied to the
+	// resource by default. It is order-sensitive.
+	Initializers []Initializer
```

Comment:
Interesting.  I would have traded-off in the other direction.  Betting that I'd have fewer `ResourceInitializers` if I did one per `Initializer` instead of one per `Resource`.  Doing it that way would also make it easy/obvious that the initializers were always wired in the same order instead of hoping that every resource puts the initializers in the same/right order.

@smarterclayton Was this already litigated in the community pull?